web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java

```diff
@@ -115,6 +129,15 @@ public void setCookieName(String cookieName) {
 		this.cookieName = cookieName;
 	}
 
+	/**
+	 * Sets the HttpOnly attribute on the cookie containing the CSRF token. Defaults to true.
+	 *
+	 * @param cookieHttpOnly true sets the HttpOnly attribute, false does not set it.
+	 */
+	public void setCookieHttpOnly(boolean cookieHttpOnly) {
```

Comment:
We should check to see a user explicitly sets `true` and `setHttpOnlyMethod ` is null. If so, we should throw an `IllegalArgumentException` since we will not be setting the cookie as HttpOnly